### PR TITLE
Add techmix per sector plot

### PR DESCRIPTION
**Pull request by MonikaFu**
Add techmix_sector function to the dashboard together with the plot data.

Note that the case of empty data selection is not handled elegantly at this moment. To be handled at a later stage. See #40.

### CODE REVIEW

File: src/routes/sector_view.svelte [hunk 1]
```diff
@@ -211,33 +217,33 @@
 				<label class="label">
 					<span>Scenario source</span>
 					<select class="select variant-outline-surface" id="scenario_source_selector">
-						<option value="GECO2023">GECO2023</option>
 						<option value="WEO2023">WEO2023</option>
+						<option value="GECO2023">GECO2023</option>
```
Comment (MonikaFu) [inline src/routes/sector_view.svelte:221]: changing order to scenario source for which techmix appears.

**Comment by jdhoffa** [comment]
* Pulled locally
* Built using `vite build`
* Previewed using `vite preview`
Interacted with the various global selectors. Beyond the issue that will be addressed in #40 , the plot looks good:
Screenshot 2024-09-23 at 13 11 11

File: src/js/techmix_sector.js [hunk 2]
```diff
@@ -0,0 +1,378 @@
+import * as d3 from 'd3';
+import { union } from 'd3-array';
+import '../css/plot_styles.css';
+
+export class techmix_sector {
+	constructor(container, data) {
+		// Data needs to be ordered on year (increasing)
+		// Data needs to have two years in
```
Comment (jdhoffa) [inline src/js/techmix_sector.js:8]: Comment seems to be missing crucial info?

File: src/js/techmix_sector.js [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+				uniqueValueTypes[idx],
+				item
+			);
+			subdataTechPerYear[idx]['greenBars'] = getGreenBarsValsPerYear(
+				subdata,
+				uniqueValueTypes[idx],
+				item
+			);
+		});
+		// TODO: make sure that this captures all possible technologies
```
Comment (jdhoffa) [inline src/js/techmix_sector.js:126]: Do we need to handle this TODO?

**Review by jdhoffa — CHANGES_REQUESTED** [review]
A few NIT comments, but generally looks decent :-) 

I think only the `TODO` would be blocking for now.

File: src/js/techmix_sector.js (hunk 2 above)
Comment (MonikaFu) [reply src/js/techmix_sector.js:8]: What do you mean? These are comments I put in about data requirements for this function. They are not exhaustive though. During documenting/refactoring stage we should probably put in tests to check if all the conditions for data are met. At the same time this function is not meant to be used by general users like functions in R packages so I am not sure to what level of detail we should document it.

Comment (jdhoffa) [reply src/js/techmix_sector.js:8]: "Data needs to have two years in" seems like it isn't a complete sentence. Two years in ... what?

Comment (MonikaFu) [reply src/js/techmix_sector.js:8]: forgive my English :D I'll make it clearer

File: src/js/techmix_sector.js [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+		);
+
+		// Declare the colours
+		const color = d3
+			.scaleOrdinal()
+			.domain(subgroups)
+			.range(d3.schemeSpectral[subgroups.length])
+			.unknown('#ccc');
+
+		// Add rectangles for each stacked bar - TODO: rewrite into two
```
Comment (jdhoffa) [inline src/js/techmix_sector.js:169]: NIT: @MonikaFu in general, we should aim to NOT have any TODOs on `main`. So my suggestion would be to please review all remaining TODOs in the comments and either:
* handle them (if directly relevant to this particular PR)
OR
* open a new GH issue explaining what should be done

**Review by jdhoffa — CHANGES_REQUESTED** [review]

**Review by jdhoffa — APPROVED** [review]

**Pull request merged**